observe/datadog.go

```diff
@@ -0,0 +1,96 @@
+package observe
+
+import (
+	datadog "github.com/DataDog/opencensus-go-exporter-datadog"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/pkg/errors"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+)
+
+// DatadogExporterConfig provides configuration for the Datadog exporter.
+// Config can be initiated with envconfig from environment
+type DatadogExporterConfig struct {
+	DatadogExporterEnabled        bool   `default:"false" split_words:"true"`
+	DatadogExporterMetricsAddress string `split_words:"true"`
+	DatadogExporterTracesAddress  string `split_words:"true"`
+	DatadogExporterNamespace      string `default:"opencensus" split_words:"true"`
+}
+
+// RegisterAndObserveDatadog will initiate and register Datadog metrics
+// and tracing exporters
+func RegisterAndObserveDatadog(onError func(error)) error {
+	if SkipObserve() {
+		return nil
+	}
+
+	var config DatadogExporterConfig
+	envconfig.Process("", &config)
+
+	exp, err := NewDatadogExporter(config, onError)
+	if err != nil {
+		return errors.Wrap(err, "unable to initiate Datadog's opencensus exporter")
+	}
+	trace.RegisterExporter(exp)
+	view.RegisterExporter(exp)
+
+	return nil
+}
+
+// NewDatadogExporter will return Datadog's opencensus exporter.
+// Exporter will send metrics and traces to Datadog's agent using
+// addresses specified through DatadogExporterConfig
+func NewDatadogExporter(config DatadogExporterConfig, onErr func(error)) (*datadog.Exporter, error) {
+
+	if !config.DatadogExporterEnabled {
+		return nil, errors.New("Datadog exporter disabled")
```

Comment:
why return an error here? 
I think you can simply return nil in line 29, no?

Reply:
I want to provide a message when this is being called but not initiated. But you're right that the error is probably too much. I'll update this with a log message.